### DISCUSSION TITLE
New features in main branch vs Hive 3 branch

### DISCUSSION THREAD

**Discussion by jmnunezizu** [ExpediaGroup/waggle-dance#319] [Ideas]
Hi folks,

In the last few months, we added two new features that we believe can also be leveraged in the hive3.x branch. These are:

- **Read-only remote metastore (optional)** – allows "read" operations to be routed to another metastore. This is useful for splitting traffic between read-write and read-only, and reducing the load in the read-write instance.
- **Rate limiting (optional)** – allows Waggle Dance to introduce throttling based on UGIs. Hive 2.x does not support rate limiting and we've experienced problems (think noisy neighbour problem).

The hive3.x branch should merge these changes to prevent it from deviating further. 

@flaming-archer – Is this something you can own?

Regards,
Jose.-

**Comment by flaming-archer** [comment]
Okay, I'll give it a try.

Besides, can this project introduce some committers/PMC mechanism. It can attract more people to participate.
For example, give me a committer firstly, hahahha. 😄😄😄😄。

**Comment by jmnunezizu** [comment]
Yes, we can introduce that. Committer privileges must be earned! ;)

We should organise a quick sync to better understand your intention/needs with Waggle Dance. We can share ours too. This will greatly simplify review time.

**Comment by flaming-archer** [comment]
Ok. A quick sync refers to whether you plan to organize a remote meeting？ I'm okay, but my spoken English is not very good, but my colleagues can.

**Reply by jmnunezizu** [reply]
Yes, that's what I meant by a quick sync. Don't worry about your English, we'll make it work!

**Reply by flaming-archer** [reply]
> Yes, that's what I meant by a quick sync. Don't worry about your English, we'll make it work!

Ok, no problem.
